integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/BaseFhirR3IntegrationTest.java

```diff
@@ -112,4 +117,98 @@ protected void describeMismatchSafely(Bundle.BundleEntryComponent item, Descript
 			matcher.describeMismatch(item.getResource(), mismatchDescription);
 		}
 	}
+	
+	protected static Matcher<List<Bundle.BundleEntryComponent>> isSortedAndWithinMax(Integer max) {
+		return new IsSortedAndWithinMax(max);
+	}
+	
+	private static class IsSortedAndWithinMax extends TypeSafeDiagnosingMatcher<List<Bundle.BundleEntryComponent>> {
+		
+		private int max;
+		
+		IsSortedAndWithinMax(int max) {
+			this.max = max;
+		}
+		
+		@Override
+		protected boolean matchesSafely(List<Bundle.BundleEntryComponent> entries, Description mismatchDescription) {
+			List<Observation> observations = entries.stream().map(Bundle.BundleEntryComponent::getResource)
+			        .filter(it -> it instanceof Observation).map(it -> (Observation) it).collect(Collectors.toList());
+			
+			Set<String> codeList = new HashSet<>();
+			
+			for (int var = 0; var < observations.size(); var++) {
+				String currentConcept = observations.get(var).getCode().getCodingFirstRep().getCode();
+				
+				//check if Bundle is returned grouped code wise
+				// if an observation arrives whose concept wise list is already created, then that means this observation is
+				//out of place
+				//so bundle is not returned grouped wise
+				if (codeList.contains(currentConcept)) {
+					mismatchDescription.appendText("an Observation which was not grouped correctly")
+					        .appendValue(observations.get(var).getId());
+					return false;
```

Comment:
In this instance, it's probably a good idea to also report which code that observation had.